wgpu-hal/src/vulkan/device.rs

```diff
@@ -604,13 +604,22 @@ impl super::Device {
     /// # Safety
     ///
     /// - `vk_image` must be created respecting `desc`
+    /// - If [`TextureUses::EXTERNAL`](crate::TextureUses::EXTERNAL) is set, then `external_queue_family_index` must be set.
     /// - If `drop_guard` is `Some`, the application must manually destroy the image handle. This
     ///   can be done inside the `Drop` impl of `drop_guard`.
     pub unsafe fn texture_from_raw(
         vk_image: vk::Image,
         desc: &crate::TextureDescriptor,
+        external_queue_family_index: Option<u32>,
         drop_guard: Option<super::DropGuard>,
     ) -> super::Texture {
+        #[cfg(debug_assertions)]
+        {
+            if desc.usage.contains(crate::TextureUses::EXTERNAL) {
+                assert!(external_queue_family_index.is_some(), "Texture has TextureUse::EXTERNAL, but does not specify the owning queue family");
```

Comment:
Let's use strict_assert! for this and remove the external debug_assertions (if the assert isn't there - this is an empty branch and will be removed)

Reply:
strict_assert is actually in wgpu-core. I can copy it over though

Reply:
Yeah, that's fine.

Reply:
THis should still use the strict-assert macro not a manual cfg